docs/src/developer/design/user_api.rst

```diff
@@ -0,0 +1,154 @@
+.. _designing_cmaizes_user_api:
+
+###########################
+Designing CMaize's User API
+###########################
+
+:ref:`overview_of_cmaizes_design` called for CMaize to have a functional-style
+user :term:`API` written over top of the object-oriented core. This page
+describes the design of CMaize's user API.
+
+TODO: Complete this page. Current content was originally part of overview, but
+then I decided to break it off.
+
+***********************
+User API Considerations
+***********************
+
+.. _functional_style_and_cmake_based:
+
+functional-style and cmake-based
+   Stemming from design discussions :ref:`overview_of_cmaizes_design` it was
+   decided that the user-facing API of CMaize needed to be written in
+   traditional CMake and should assume functional-style programming.
+
+Expected Workflow Considerations
+================================
+
+The next set of considerations, in order, represent the major steps we expect
+a CMaize-based build system to take. Generally speaking these
+considerations directly address what CMake should do in each
+of the :term:`build phase`\ s. That said CMake will parse the entire
+build system during the :ref:`configure_phase`, even the components applying to
+later phases.
+
+.. _project_meta_data:
+
+project meta data
+   Following from the :ref:`functional_style_and_cmake_based` consideration the
+   build system the user writes with CMaize should be pure CMake and invoked by
+   CMake. CMake requires (technically strongly suggests) that the first things
```

Comment:
```suggestion
   CMake. CMake requires that the first things
```
I don't think this technicality is important to include unless you think someone is going to call us out for it. By including it, I think it raises more questions instead of helping the user understand the workflow.